### ISSUE TITLE
Feature Request: Semantic Highlighting Support

### ISSUE THREAD

**Issue by ghost** [saem/vscode-nim#22]
Nimsuggest supports semantic highlighting through which it has been implemented in the nvim plugin https://github.com/alaviss/nim.nvim. Request to implement it for vscode :D

**Comment by saem** [comment]
I've thought about it, I've been rather involved with nimsuggest and there is a bug or two with the syntax based highlighting that I'd want to get to beforehand. 

I won't say no to a PR or if you wanted some guidance getting started and along the way -- via comments and/or even voice chat.

It's a good idea.

**Comment by zetashift** [comment]
https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide 
https://github.com/microsoft/vscode-extension-samples/tree/master/semantic-tokens-sample

Could these links be helpful incorporating semantic highlighting?

**Comment by saem** [comment]
Yup, that's the right idea. The bridge between nimsuggest and the editor is there, a few FFI declarations for VSCode API and plumbing code could do it.

**Comment by doongjohn** [comment]
any update?
